bin/debian.go

```diff
@@ -326,30 +340,26 @@ func doClientDeb() error {
 		}
 	}
 
-	fd, err := os.Open(input)
+	e, err := elf.Open(input)
 	if err != nil {
-		return fmt.Errorf("Unable to open executable: %w", err)
+		return fmt.Errorf("Unable to parse ELF executable: %w", err)
 	}
-	defer fd.Close()
 
-	header := make([]byte, 4)
-	_, err = fd.Read(header)
-	if err != nil {
-		return fmt.Errorf("Unable to open executable: %w", err)
-	}
-
-	if binary.LittleEndian.Uint32(header) != 0x464c457f {
-		return fmt.Errorf("Binary does not appear to be an " +
```

Comment:
Maybe add this back

Reply:
I removed this check as it is obsoleted by the more comprehensive sanity check performed by `elf.Open(input)`.

Reply:
Yeah but the issue is that for the user who is trying to build an RPM from eg a windows machine they may not know why it does not work - in that case it requires the --binary flag to be provided.

Reply:
The check is fine but the error message should remain.

Reply:
Good call - the previous error message was more actionable. I've updated the PR to an error message that should balance actionable & informative.

Here's the new error when you pass in a Windows binary:

```log
artifacts/testdata/files/notnbt.exe is not a valid Linux ELF binary. Use the --binary flag to specify the
 path to a Linux binary: bad magic number '[77 90 144 0]' in record at byte 0x0 
```

Here's the new error when you pass in an empty file:

```log
./empty is not a valid Linux ELF binary. Use the --binary flag to specify the path to a Linux binary: EOF
```

Here's the new error when you pass in a non-existent file:

```log
/missing/path is not a valid Linux ELF binary. Use the --binary flag to specify the path to a Linux binar
y: open /missing/path: no such file or directory
```

Thoughts?